### PR TITLE
[Matlab] Add code section comments to symbol list

### PR DESCRIPTION
**Pull request by jwortmann**
Reference from the Matlab docs: https://www.mathworks.com/help/matlab/matlab_prog/create-and-run-sections.html

The scope `entity.name.section` is added to the section title, which is already used in R and gets a "navigation" kind icon in the Goto Symbol panel. Also the pattern was adjusted so that section comments should start on their own line.

### CODE REVIEW

File: Matlab/Matlab.sublime-syntax [hunk 1]
```diff
@@ -467,10 +467,11 @@ contexts:
     - include: block-comments
 
   line-comments:
-    - match: (%%(?!%)).*\n?
+    - match: ^\s*(%%(?!%))\s*(.*\S)?\s*\n?
```
Comment (jfcherng) [inline Matlab/Matlab.sublime-syntax:470]: is `^` necessary?

Comment (jwortmann) [reply Matlab/Matlab.sublime-syntax:470]: According to the link above
> You also can enter two percent signs (%%) at the start of the line where you want to begin the new section.

(but they also work after whitespace nested within e.g. loops to create section breaks)

**Comment by jrappen** [comment]
Based on your changes to the test file and that passing (without checking further), you probably:

- now match the second rule in `line-comments` with the test
- which doesn't have a capture for `entity.*`
- which makes the negative test redundant
- which makes me think you messed this up

Maybe double-check `comment.line.percentage` vs. `comment.line.double-percentage`?

**Comment by jwortmann** [comment]
The negative test for `entity` is redundant, yes, but it doesn't hurt and it makes it clear that we don't want to match it as a code section comment if there is anything other than whitespace on the same line before the comment (unlike the new highlighting here on GitHub).

**Review by deathaxe — APPROVED** [review]

**Review by jfcherng — APPROVED** [review]

**Comment by deathaxe** [comment]
> Maybe double-check `comment.line.percentage` vs. `comment.line.double-percentage`?

No matter how many leading percentages a comment has only section comments are of special meaning. That's exactly the strategy used in #3217 to limit the amount of different comment scopes.

**Comment by jwortmann** [comment]
I noticed that the adjusted pattern for section comments accidentally resulted in applying the `comment` scope to preceding whitespace as well. This was not intended, but on the other hand it makes it possible to look a bit like in the Matlab IDE, by using an adjusted background color for that scope in the color scheme:

matlab

matlab2

Should it be "fixed", or leave it as it is? (The fix would be to add a capture group and move `comment.line.double-percentage.matlab` from "scope" to "captures".)

**Comment by deathaxe** [comment]
I think it's ok this way.

**Pull request merged**